pkg/controller/sidecarset/sidecarset_processor.go

```diff
@@ -103,6 +103,13 @@ func (p *Processor) UpdateSidecarSet(sidecarSet *appsv1alpha1.SidecarSet) (recon
 		return reconcile.Result{RequeueAfter: time.Second}, nil
 	}
 
+	// check if pod finish sidecar container upgrade
+	for _, pod := range pods {
```

Comment:
Maybe we should make `updatePodsSidecarHashState` after [line-99](https://github.com/openkruise/kruise/pull/1349/files#diff-0044cb47203edfb25c4c6e701614e275692fd84828140d452b3692b1106ae5bbR99) because some pods that already ready should be update state to `Noamal`.